rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +270,302 @@ def historical_uniswap_v2_lp_price(
+            ],
+        )
+        slots_0 = [
+            entry[0].decode(entry[1][1], 'slot0')
+            for entry in zip(pool_contracts, slots_0_multicall)
+        ]
+        tokens_a = [
+            ethereum.get_basic_contract_info(to_checksum_address(position[2]))
+            for position in positions
+        ]
+        tokens_b = [
+            ethereum.get_basic_contract_info(to_checksum_address(position[3]))
+            for position in positions
+        ]
+        price_ranges = [
+            calculate_price_range(
+                tick_lower=entry[2][5],
+                tick_upper=entry[2][6],
+                decimal_0=entry[0]['decimals'],
+                decimal_1=entry[1]['decimals'],
+            )
+            for entry in zip(tokens_a, tokens_b, positions)
+        ]
+        amounts_0 = [
+            calculate_amount_0(
+                tick=entry[0][1],
+                tick_bottom=entry[1][5],
+                tick_top=entry[1][6],
+                liquidity=entry[1][7],
+                decimals=entry[2]['decimals'],
+            )
+            for entry in zip(slots_0, positions, tokens_a)
+        ]
+        amounts_1 = [
+            calculate_amount_1(
+                tick=entry[0][1],
+                tick_bottom=entry[1][5],
+                tick_top=entry[1][6],
+                liquidity=entry[1][7],
+                decimals=entry[2]['decimals'],
+            )
+            for entry in zip(slots_0, positions, tokens_b)
+        ]
+        total_tokens_in_pools = []
+        if premium:
+            liquidity_in_pools_multicall = multicall_2(
+                ethereum=ethereum,
+                require_success=True,
+                calls=[
+                    (entry[0], entry[1].encode('liquidity'))
+                    for entry in zip(pool_addresses, pool_contracts)
+                ],
+            )
+            liquidity_in_pools = [
+                (entry[0].decode(entry[1][1], 'liquidity')[0])
+                for entry in zip(pool_contracts, liquidity_in_pools_multicall)
+            ]
+            for entry in zip(slots_0, liquidity_in_pools, positions, tokens_a, tokens_b):
+                total_tokens_in_pools.append(
+                    calculate_total_amounts_of_tokens(
+                        liquidity=entry[1],
+                        tick=entry[0][1],
+                        fee=entry[2][4],
+                        decimal_0=entry[3]['decimals'],
+                        decimal_1=entry[4]['decimals'],
+                    ),
+                )
+        for entry in itertools.zip_longest(
+            tokens_ids,
+            pool_addresses,
+            positions,
+            price_ranges,
+            tokens_a,
+            tokens_b,
+            amounts_0,
+            amounts_1,
+            total_tokens_in_pools,
+            fillvalue=(None, None),
+        ):
+            if entry[6] > ZERO or entry[7] > ZERO:  # type: ignore
```

Comment:
Why this type ignore here? Why not converting the entry items to FVal to compare?

Reply:
The `# type: ignore` is due to mypy returning the error below when accessing the tuple element even though the tuple is not out of range.
> rotkehlchen/chain/ethereum/modules/uniswap/utils.py:470: error: Tuple index out of range